gax/src/test/java/com/google/api/gax/batching/v2/BatcherImplTest.java

```diff
@@ -0,0 +1,197 @@
+  public void testBatchingSuccess() throws Exception {
+    when(mockDescriptor.newRequestBuilder(any(LabeledIntList.class)))
+        .thenReturn(SQUARER_BATCHING_DESC_V2.newRequestBuilder(labeledIntList));
+    when(unaryCallable.futureCall(any(LabeledIntList.class)))
+        .thenReturn(ApiFutures.immediateFuture(Collections.singletonList(16)));
+
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) {
+                List<Integer> rpcResponse = invocation.getArgument(0);
+                List<SettableApiFuture<Integer>> apiFutures = invocation.getArgument(1);
+                apiFutures.get(0).set(rpcResponse.get(0));
+                return null;
+              }
+            })
+        .when(mockDescriptor)
+        .splitResponse(Mockito.<Integer>anyList(), Mockito.<SettableApiFuture<Integer>>anyList());
+
+    ApiFuture<Integer> result = underTest.add(4);
+    underTest.flush();
+    assertThat(result.isDone()).isTrue();
+    assertThat(result.get()).isEqualTo(16);
+
+    verify(mockDescriptor, times(1)).newRequestBuilder(labeledIntList);
+    verify(mockDescriptor, times(1))
+        .splitResponse(Mockito.<Integer>anyList(), Mockito.<SettableApiFuture<Integer>>anyList());
+    verify(unaryCallable, times(1)).futureCall(any(LabeledIntList.class));
+  }
+
+  /** Verifies exception occurred at RPC is propagated to element results */
+  @Test(expected = ExecutionException.class)
+  public void testBatchingFailed() throws Exception {
+    final Exception exception = new RuntimeException();
+    when(mockDescriptor.newRequestBuilder(any(LabeledIntList.class)))
+        .thenReturn(SQUARER_BATCHING_DESC_V2.newRequestBuilder(labeledIntList));
+    when(unaryCallable.futureCall(any(LabeledIntList.class)))
+        .thenReturn(ApiFutures.<List<Integer>>immediateFailedFuture(exception));
+    doAnswer(
+            new Answer<Void>() {
+              @Override
+              public Void answer(InvocationOnMock invocation) {
+                List<SettableApiFuture<Integer>> apiFutures = invocation.getArgument(1);
+                apiFutures.get(0).setException(exception);
+                return null;
+              }
+            })
+        .when(mockDescriptor)
+        .splitException(any(Throwable.class), Mockito.<SettableApiFuture<Integer>>anyList());
+    ApiFuture<Integer> failedResult = underTest.add(5);
+    underTest.flush();
+    verify(mockDescriptor, times(1)).newRequestBuilder(labeledIntList);
+
+    verify(mockDescriptor, times(1))
+        .splitException(any(Throwable.class), Mockito.<SettableApiFuture<Integer>>anyList());
+    verify(unaryCallable, times(1)).futureCall(any(LabeledIntList.class));
+    assertThat(failedResult.isDone()).isTrue();
+    try {
+      failedResult.get();
+    } catch (InterruptedException | ExecutionException e) {
+      Truth.assertThat(e.getCause()).isInstanceOf(RuntimeException.class);
+      throw e;
+    }
```

Comment:
Please use something like:
https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/stub/AwaitReplicationCallableTest.java#L97-L105To test exceptions, it makes it easier to see whats happening as opposed using a combination of expected annotations and rethrowing it.

Also, please unwrap the ExecutionException and verify that the cause is expected